alerting/src/main/kotlin/org/opensearch/alerting/transport/TransportDeleteWorkflowAction.kt

```diff
@@ -0,0 +1,248 @@
+                QueryBuilders.nestedQuery(
+                    Workflow.WORKFLOW_DELEGATE_PATH,
+                    QueryBuilders.boolQuery().must(
+                        QueryBuilders.termsQuery(
+                            Workflow.WORKFLOW_MONITOR_PATH,
+                            monitorIds
+                        )
+                    ),
+                    ScoreMode.None
+                )
+            )
+
+            val searchRequest = SearchRequest()
+                .indices(ScheduledJob.SCHEDULED_JOBS_INDEX)
+                .source(SearchSourceBuilder().query(queryBuilder).fetchSource(true))
+
+            val searchResponse: SearchResponse = client.suspendUntil { search(searchRequest, it) }
+
+            val workflows = searchResponse.hits.hits.map { hit ->
+                val xcp = XContentHelper.createParser(
+                    xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                    hit.sourceRef, XContentType.JSON
+                ).also { it.nextToken() }
+                lateinit var workflow: Workflow
+                while (xcp.nextToken() != XContentParser.Token.END_OBJECT) {
+                    xcp.nextToken()
+                    when (xcp.currentName()) {
+                        "workflow" -> workflow = Workflow.parse(xcp)
+                    }
+                }
+                workflow.copy(id = hit.id, version = hit.version)
+            }
+            val workflowMonitors = workflows.filter { it.id != workflowIdToBeDeleted }.flatMap { (it.inputs[0] as CompositeInput).getMonitorIds() }.distinct()
+            // Monitors that can be deleted -> all workflow delegates - monitors belonging to different workflows
+            return monitorIds.minus(workflowMonitors.toSet())
+        }
+
+        private suspend fun getWorkflow(): Workflow {
+            val getRequest = GetRequest(ScheduledJob.SCHEDULED_JOBS_INDEX, workflowId)
+
+            val getResponse: GetResponse = client.suspendUntil { get(getRequest, it) }
+            if (getResponse.isExists == false) {
+                actionListener.onFailure(
+                    AlertingException.wrap(
+                        OpenSearchStatusException("Workflow not found.", RestStatus.NOT_FOUND)
+                    )
+                )
+            }
+            val xcp = XContentHelper.createParser(
+                xContentRegistry, LoggingDeprecationHandler.INSTANCE,
+                getResponse.sourceAsBytesRef, XContentType.JSON
+            )
+            return ScheduledJob.parse(xcp, getResponse.id, getResponse.version) as Workflow
+        }
+
+        private suspend fun deleteWorkflow(workflow: Workflow): DeleteResponse {
+            log.debug("Deleting the workflow with id ${deleteRequest.id()}")
+            return client.suspendUntil { delete(deleteRequest, it) }
+        }
+
+        private suspend fun deleteMetadata(workflow: Workflow) {
```

Comment:
plz add debug logs like "deleting metadata",

Reply:
Ok this part is copy pasted - even we don't have workflow metada. will comment out the call and add TODO - since we are going to introduce the metadata with workflow execution. Tnx!